client/src/shared/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { Fragment } from 'react';
+
+export interface ModalProps {
+  children?: any
+  isOpen: boolean
+}
+
+const Modal: React.FC<ModalProps> = ({
+  children,
+  isOpen
+}: ModalProps) => {
+  if (!isOpen) return null;

```

Comment:
There is a hook I created named useOutsideClick, you may want to use it as well so that when a user clicks anything outside the modal, it will automatically close